app/controllers/line_items_controller.rb

```diff
@@ -1,52 +1,59 @@
+# frozen_string_literal: true
+
 class LineItemsController < ApplicationController
-  before_action :find_item, only: %i[ destroy add_quantity reduce_quantity]
+  before_action :find_item, only: %i[destroy add_quantity reduce_quantity]
 
-  def create       
+  def create
     chosen_product = Product.find(params[:product_id])
     current_cart = @current_cart
     if current_cart.products.include?(chosen_product)
-      @line_item = current_cart.line_items.find_by(:product_id => chosen_product)
+      @line_item = current_cart.line_items.find_by(product_id: chosen_product)
       @line_item.quantity += 1
     else
       @line_item = LineItem.new
       @line_item.cart = current_cart
       @line_item.product = chosen_product
-      @line_item.quantity = 1      
+      @line_item.quantity = 1
+    end
+    begin
+      @line_item.save!
+      redirect_to cart_path(current_cart)
+    rescue Exception => e
+      redirect_to request.referrer, alert: e.message
     end
-    @line_item.save!
-    redirect_to cart_path(current_cart)
   end
-    
+
   def destroy
     @line_item.destroy
     redirect_to cart_path(@current_cart)
-  end  
-    
+  end
+
   def add_quantity
     @line_item.quantity ||= 0
     @line_item.quantity += 1
-    @line_item.save!
-    redirect_to cart_path(@current_cart)
+    if @line_item.update_attribute(:quantity, @line_item.quantity)
```

Comment:
use update_attributes.
Check difference between update_attribute, update_column and update_attributes

Reply:
used update instead of update_attribute